### ISSUE TITLE
Scroll to center is ignored if already visible

### ISSUE THREAD

**Issue by 314eter** [atom/atom#11429]
### Description

When scrolling to a position that is already visible with the `center` option set to `true` (as described [here](https://atom.io/docs/api/v1.6.2/TextEditor#instance-scrollToBufferPosition)), no scrolling is performed.
### Steps to Reproduce

``` javascript
editor = atom.workspace.getActiveTextEditor() /* open a file with 100 lines */
editor.scrollToBufferPosition([50, 0]) /* scroll to line 50, line 52 is visible now */
editor.scrollToBufferPosition([52, 0], {center: true})
```

**Expected behavior:** According to the API docs, line 52 should be scrolled to the center of the editor.

**Actual behavior:** Nothing happens.
### Further information

Before e3dabd5200afea5074fb63003951ef677539413e, the behavior was as expected. Maybe the change was intentional, but then I don't understand why.

The problem can be fixed by removing

``` coffeescript
unless @getScrollTop() < desiredScrollCenter < @getScrollBottom()
```

in [text-editor-presenter.coffee](https://github.com/atom/atom/blob/master/src/text-editor-presenter.coffee). I can open a PR with this change if necessary.

**Comment by Ben3eeE** [comment]
Activity

**Comment by 314eter** [comment]
The issue is still reproducible.

**Comment by rsese** [comment]
:+1:

> What version of Atom you reproduced the issue on

1.32.0-nightly6

> What OS and version you reproduced the issue on

macOS 10.12.6

> What steps you followed to reproduce the issue

The steps in the issue body.

**Comment by divieira** [comment]
Also ran into this. Seems to be related to the (closed) issues #3131 and #3324.
